game_state.go

```diff
@@ -27,6 +27,8 @@ type GameState struct {
 	lastFlash          lastFlasher    // Information about the last flash that exploded, used to find the attacker and projectile for player_blind events
 	currentDefuser     *common.Player // Player currently defusing the bomb, if any
 	currentPlanter     *common.Player // Player currently planting the bomb, if any
+	thrownGrenades      map[*common.Player]map[int]*common.Equipment // Information about every player's thrown grenades (from the moment they are thrown to the moment their effect is ended)
```

Comment:
actually, let's make this a `map[*common.Player]map[EquipmentElement]*common.Equipment`.

I noticed we don't actually need the entityID

Reply:
Actually, i use it for deletion (when HE grenade projectile get destroyed) in database.go -> nadeProjectileDestroyed -> line 443

Reply:
Couldn't we also use the `EquipmentElement` to do this?

Reply:
@markus-wa We can't use `EquipmentElement` as map key because we can have multiple grenades of the same type stored in this map for a same player.

Here's a summary example of what could happen : 
- Player1 throw Flash1
- Player1 throw Flash2 -> We now store 2 flash grenades in our map
- Flash1 explose & flash Player3 & 4 -> We pop out the Flash1 from map
- Flash2 explose & flash Player5 -> We pop out the Flash2 from map

Note: this could happen with HE (molotiv etc..)  too if a player throw a nade of this kind, pick another on the ground and throw it ASAP

However if you want to "unify" the deletion process, we might use our new deletion method here too by doing something like :  
```go
if !isInferno {
   p.gameEventHandler.deleteThrownGrenadeByType(proj.Owner, p.WeaponInstance.Weapon)
}
```
Not sure if it's better, but should work the same way i guess.

What do you think ?

Reply:
After adding some unit tests i realised it has some issues with Map initialisation.

I commited my changes with new definition :
```go
thrownGrenades      map[*common.Player][]*common.Equipment
```
It's now a **Map** of **Slices** which is make my easier to handle (I don't have to handle initialisation like doing a make for the map in the map at the right moment).

Plus i'm now always using the `deleteThrownGrenade` method !

Just resolve this if it's fine for you!

Reply:
sounds good 🙂